server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java

```diff
@@ -330,6 +330,10 @@ protected double computeCost(
     List<ListenableFuture<Pair<Double, ServerHolder>>> futures = Lists.newArrayList();
 
     for (final ServerHolder server : serverHolders) {
+      if (server.getAvailableSize() < proposalSegment.getSize()) {
```

Comment:
can this ever result in an infinite loop where the same server is selected over and over?